contracts/AaveV2StablecoinCellar.sol

```diff
@@ -199,995 +291,560 @@ contract AaveV2StablecoinCellar is IAaveV2StablecoinCellar, ERC20, Ownable {
-     *         ASSUMING they are active.
+     * @notice The amount of assets that the cellar would exchange for the amount of shares provided.
      * @param shares amount of shares to convert
      * @return assets the shares can be exchanged for
      */
-    function convertToAssets(uint256 shares) public view returns (uint256) {
-        uint256 assets = _convertToAssets(shares);
-        return assets.changeDecimals(decimals, assetDecimals);
-    }
+    function convertToAssets(uint256 shares) public view override returns (uint256 assets) {
+        uint256 totalShares = totalSupply; // Saves an extra SLOAD if totalSupply is non-zero.
+        uint8 positionDecimals = assetDecimals;
+        uint256 totalAssetsNormalized = totalAssets().changeDecimals(positionDecimals, 18);
 
-    /**
-     * @dev Same as `convertToAssets` but forcibly denoted with 18 decimals of precision.
-     */
-    function _convertToAssets(uint256 shares) internal view returns (uint256) {
-        return totalSupply == 0 ? shares : shares.mulDivDown(_totalAssets(), totalSupply);
+        assets = totalShares == 0 ? shares : shares.mulDivDown(totalAssetsNormalized, totalShares);
+        assets = assets.changeDecimals(18, positionDecimals);
     }
 
     /**
-    * @notice Simulate the effects of depositing assets at the current block, given current on-chain
-    *         conditions.
-     * @param assets amount of assets to deposit
-     * @return shares that will be minted
+     * @notice The amount of shares that the cellar would exchange for the amount of assets provided.
+     * @param assets amount of assets to convert
+     * @return shares the assets can be exchanged for
      */
-    function previewDeposit(uint256 assets) public view returns (uint256) {
-        return convertToShares(assets);
+    function convertToShares(uint256 assets) public view override returns (uint256 shares) {
+        uint256 totalShares = totalSupply; // Saves an extra SLOAD if totalSupply is non-zero.
+        uint8 positionDecimals = assetDecimals;
+        uint256 assetsNormalized = assets.changeDecimals(positionDecimals, 18);
+        uint256 totalAssetsNormalized = totalAssets().changeDecimals(positionDecimals, 18);
+
+        shares = totalShares == 0 ? assetsNormalized : assetsNormalized.mulDivDown(totalShares, totalAssetsNormalized);
     }
 
     /**
-    * @notice Simulate the effects of minting shares at the current block, given current on-chain
-    *         conditions.
+     * @notice Simulate the effects of minting shares at the current block, given current on-chain conditions.
      * @param shares amount of shares to mint
      * @return assets that will be deposited
      */
-    function previewMint(uint256 shares) public view returns (uint256) {
-        uint256 supply = totalSupply; // Saves an extra SLOAD if totalSupply is non-zero.
+    function previewMint(uint256 shares) public view override returns (uint256 assets) {
```

Comment:
Do we need the same logic for `previewDeposit`, or alternatively, do we need to override `deposit` from ERC4626 to disable it? I don't see how deposit is doing the same decimal accounting as `previewMint` does here